Digital Twin Manager 1025.2.1 Changelogs for UI 1025.0.4 (#4604)
SoftwareAG/c8y-docs: content/change-logs/application-enablement/dtm-1025-2-1-use-inventory-api-for-assets-listing-screen-to-fix-pagination-and-row-expansion.md

---
date: ""

> title: "Fix frozen pagination and row expansion in Assets page"

Reviewer: ```suggestion
title: "Pagination and row expansion in Assets page no longer frozen"
```

product_area: "Application enablement & solutions"
change_type:
    - value: "change-VSkj2iV9m"
      label: "Fix"
component:
    - value: "component-Tl88RYb4A"
      label: "Digital Twin Manager"
build_artifact:
    - value: "tc-wYIY0MBDO"
      label: "dtm"
ticket: "CTM-2895"
version: "1025.2.1"
---

> Previously, the pagination and row expansion froze in the Assets page when displaying large datasets. This is now resolved to provide a reliable pagination and row expansion.

Reviewer: ```suggestion
Previously, the pagination and row expansion in the **Assets** page were frozen when displaying large datasets. This issue is now resolved to provide a reliable pagination and row expansion.
```

---
date: ""

> title: "Use Inventory API for assets listing screen to fix pagination and row expansion."

Reviewer 2: ```suggestion
title: "Fix frozen pagination and row expansion in Assets page"
```

product_area: "Application enablement & solutions"
change_type:
    - value: "change-VSkj2iV9m"
      label: "Fix"
component:
    - value: "component-Tl88RYb4A"
      label: "Digital Twin Manager"
build_artifact:
    - value: "tc-wYIY0MBDO"
      label: "dtm"
ticket: "CTM-2895"
version: "1025.2.1"
---
The Assets screen in Digital Twin Manager experienced pagination and
row-expansion issues when displaying large datasets. The DTM Assets list
API failed when **withSubAssets: true** was set, which caused issues. As
a workaround, the Assets screen now uses the Cumulocity Inventory API
instead of the DTM Assets API for fetching root-level assets and
asset-by-type queries. This provides reliable pagination and correct

> row-expansion behavior.

Reviewer 2: ```suggestion
Previously, the pagination and row expansion froze in the Assets page when displaying large datasets. This is now resolved to provide a reliable pagination and row expansion.
```
